cmd/adsysd/integration_tests/adsys_mount_test.go

```diff
@@ -0,0 +1,108 @@
+package adsys_test
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestUserMountHandler(t *testing.T) {
+	fixtureDir := filepath.Join("testdata", t.Name())
+
+	env, target := setupBinaryForTests(t)
+
+	tests := map[string]struct {
+		mountsFile    string
+		sessionAnswer string
+		noKrbTicket   bool
+
+		wantStatus int
+	}{
+		// Single entries
+		"mount successfully nfs share": {mountsFile: "mounts_with_nfs_entry"},
+		"mount successfully smb share": {mountsFile: "mounts_with_smb_entry"},
+
+		// Anonymous entries
+		"mount successfully one anonymous nfs entry": {mountsFile: "mounts_with_anonymous_nfs_entry"},
+		"mount successfully one anonymous smb entry": {mountsFile: "mounts_with_anonymous_smb_entry"},
+
+		// Many entries
+		"mount successfully many entries with different protocols": {mountsFile: "mounts_with_many_entries"},
+		"mount successfully many anonymous entries":                {mountsFile: "mounts_with_many_anonymous_entries"},
+
+		// File cases
+		"exit code 0 when file is empty": {mountsFile: "mounts_with_no_entries"},
+
+		// File errors
+		"error when file has badly formated entries": {mountsFile: "mounts_with_bad_entries", wantStatus: 1},
+		"error when file doesn't exist":              {mountsFile: "do_not_exist", wantStatus: 1},
+
+		// Authentication errors
+		"error when trying to mount smb without kerberos ticket":   {mountsFile: "mounts_with_smb_entry", noKrbTicket: true, wantStatus: 1},
+		"error when trying to mount nfs without kerberos ticket":   {mountsFile: "mounts_with_nfs_entry", noKrbTicket: true, wantStatus: 1},
+		"error when anonymous auth is not supported by the server": {mountsFile: "mounts_with_anonymous_smb_entry", sessionAnswer: "anonymous_error", noKrbTicket: true, wantStatus: 1},
```

Comment:
I think we can have one more case (above), like "mount successfully with anynomous auth without kerberos ticket", so that this completes those tests.